apps/api/stubs/onelogin/saml2/auth.pyi

```diff
@@ -0,0 +1,23 @@
+from onelogin.saml2.settings import OneLogin_Saml2_Settings
+
+class OneLogin_Saml2_Auth:
+    def __init__(
+        self,
+        request_data: dict[str, object],
+        old_settings: OneLogin_Saml2_Settings | dict[str, object] | None = ...,
+        custom_base_path: str | None = ...,
+    ) -> None: ...
+    def process_response(self, request_id: str | None = None) -> None: ...
```

Comment:
Thought [non-blocking]: Would it also work to use `Optional[str]` in these stub files?

Reply:
The union syntax was generated by stubgen, and I tend to follow FastAPI's advice on [Using `Union` or `Optional`](https://fastapi.tiangolo.com/python-types/#using-union-or-optional)
> Both are equivalent and underneath they are the same, but [FastAPI] would recommend `Union` instead of `Optional` because the word "**optional**" would seem to imply that the value is optional, and it actually means "it can be `None`", even if it's not optional and is still required.

Open to suggestions/discussions otherwise though. I am also surprised the new Python 3.10 syntax is allowed in stubs when running with Python 3.9.